terminal/src/main/java/org/jline/terminal/TerminalBuilder.java

```diff
@@ -260,6 +261,11 @@ public TerminalBuilder paused(boolean paused) {
     }
 
     public Terminal build() throws IOException {
+        if (TERMINAL_HOLDER.get() != null) {
+             Terminal terminal = TERMINAL_HOLDER.get();
```

Comment:
Given TERMINAL_HOLDER is an AtomicReference, I'd prefer if you could reuse the `terminal` variable defined below and use a single call to avoid any _theoric_ threading issue. 
```
Terminal terminal = TERMINAL_HOLDER.get();
if (terminal != null) {
   ...
}
terminal = doBuild();
```

Reply:
No problem. That was my original idea but I did it this way to avoid merge conflicts in the case where we were maintaining a fork.

Reply:
I also renamed `TERMINAL_HOLDER` to `TERMINAL_OVERRIDE` to be more consistent with `SYSTEM_TERMINAL`.